sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp

```diff
@@ -104,6 +104,12 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
        * @brief Holds the customer provided key used when making requests.
        */
       Azure::Nullable<EncryptionKey> CustomerProvidedKey;
+
+      /**
+       * The filesystem url. This is only non-null for directory clients that are created from a
+       * filesystem client, so that this directory client knows where to send ListPaths requests.
+       */
+      Azure::Nullable<Azure::Core::Url> FileSystemUrl;
```

Comment:
In general, `ClientOptions` are input-only and are immutable, but the comment implies that this is a value set by the FileSystemClient.

What happens when a customer sets this value? Do you REALLY need to expose this to external customers?